src/coreComponents/dataRepository/Group.hpp

```diff
@@ -1712,6 +1731,24 @@ Wrapper< T > & Group::registerWrapper( string const & name,
   return rval;
 }
 
+template< typename LOG_LEVEL_INFO >
+std::enable_if_t< geos::is_log_level_info< LOG_LEVEL_INFO >, void >
+Group::addLogLevel()
+{
+  GEOS_ERROR_IF( m_logLevelsRegistry == nullptr, "You cannot call addLogLevel after schema generation" );
+
+  Wrapper< integer > * wrapper = getWrapperPointer< integer >( viewKeyStruct::logLevelString() );
```

Comment:
```suggestion
  WrapperBase * wrapperBase = getWrapperPointer< integer >( viewKeyStruct::logLevelString() );
```

there was a compiler error I ran into about a failed cast in a situation like this. The typed `Wrapper` was not required, so we can just use a `WrapperBase`. We could add a `getWrapperBasePointer`

Reply:
In your 1st remark, i should write smth like this ? : 
```
WrapperBase * wrapper = getWrapperPointer< integer >( viewKeyStruct::logLevelString() );
 Wrapper< integer > & wrapperT = Wrapper< integer >::cast( *wrapper );
```

Reply:
I don't understand, with your example how can I use the wrapper method (i.e setApplyDefaultValue), I need to cast to wrapper<integer> ? or I miss smth ?

Reply:
no. I didn't see you were calling `setApplyDefaultValue()` for some reason. disregard.